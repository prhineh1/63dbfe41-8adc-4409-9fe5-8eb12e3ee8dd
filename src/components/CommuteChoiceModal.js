import React from 'react';
import Modal from 'react-modal';
import moment from 'moment';
import ReactAnimatedWeather from 'react-animated-weather';

const CommuteChoiceModal = (props) => {
  const modeDeclaration = (
    props.weather.mode === 'metro' ? 'You should take the metro.' :
    "You're all clear; bike it."
  );
  return (
    <Modal
      className='modal'
      isOpen={props.isOpen} // propTypes.bool.isRequired
      onRequestClose={props.toggleModal} //propTypes.func
      contentLabel='D.C. Weather and Your Commute' // adds aria-label to the modal
      closeTimeoutMS={200} // for transition effect
    >
      <h2 className='modal__message'>{modeDeclaration}</h2>
      <div className='modal__weather-icon'>
        <ReactAnimatedWeather
          icon={props.weather.icon}
          size={96}
        />
      </div>
      <div className='modal__forecast'> 
        <span className='model__forecast__text'>Forecast for Washington D.C.</span>       
        {/* See https://momentjs.com/docs/#/displaying/ for display formatting of momentObj */}
        <span className='model__forecast__text'>{moment(props.weather.commuteDate * 1000).format('Do MMM, YYYY [at] h:mm A')}</span>
        <span className='model__forecast__text'>{props.weather.summary}</span>  
        <span className='model__forecast__text'>{props.weather.temperature} &#8457;</span>
        <span className='model__forecast__text'>Chance of precipitation: {props.weather.precipProbability * 100}%</span>
        <button className='button button--modal' onClick={props.toggleModal}>Ok</button>
      </div>
    </Modal>
  )
}

export default CommuteChoiceModal;
