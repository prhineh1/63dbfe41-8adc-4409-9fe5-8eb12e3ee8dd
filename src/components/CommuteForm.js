import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import Timepicker from 'rc-time-picker';

export default class CommuteForm extends React.Component {
  state = {
    tempHi: '',
    tempLow: '',
    precipProbability: '',
    commuteDay: moment(),
    commuteTime: moment(),
    calendarFocused: false,
    error: '',
  };
  onLowChange = e => {
    const tempLow = e.target.value;
    if (!tempLow || tempLow.match(/^[1-9][0-9]{0,2}?$/)) { //matches any number between 0 and 1000
      this.setState(() => ({ tempLow }));
    }
  };
  onHighChange = e => {
    const tempHi = e.target.value;
    if (!tempHi || tempHi.match(/^[1-9][0-9]{0,2}?$/)) {
      this.setState(() => ({ tempHi }));
    }
  };
  onPrecipChange = e => {
    const precipProbability = e.target.value;
    if (!precipProbability || precipProbability.match(/(^[1-9][0-9]?$)|100?$|^0?$/)) { //matches any number between 0 and 100
      this.setState(() => ({ precipProbability }))
    }
  }
  onDayChange = commuteDay => {
    if (commuteDay) {
      this.setState(() => ({ commuteDay }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  isOutsideRange = day => !day.isBetween(moment().subtract(1, 'd'), moment().add(14, 'd')); // Can only select dates within two weeks of today
  onTimeChange = commuteTime => {
    if (commuteTime) {
      this.setState(() => ({ commuteTime }));      
    }
  };
  onSubmit = e => {
    e.preventDefault();

    if(!this.state.tempHi || !this.state.tempLow || !this.state.precipProbability) {
      this.setState(() => ({ error: 'Please provide values for the first three fields.' }))
    } else if (parseInt(this.state.tempLow) > parseInt(this.state.tempHi)) {
      this.setState(() => ({ error: "The Hi temperature (second input) must be greater than the low temperature (first input)." }))
    } else {
      this.setState(() => ({ error: '' }));

      // create a momentObj combining the year, month and date of commuteDay
      // with the hour from commuteTime to get the appropriate commuteDate
      const commuteDate = moment({  
        year: this.state.commuteDay.year(),
        month: this.state.commuteDay.month(),
        date: this.state.commuteDay.date(),
        hour: this.state.commuteTime.hour()
      });
      this.props.onSubmit({
        tempLow: parseInt(this.state.tempLow),
        tempHi: parseInt(this.state.tempHi),
        precipProbability: parseFloat(this.state.precipProbability) / 100,
        commuteDate: commuteDate.unix() // convert momentObj to Unix timestamp for use with DarkSky API
      });
    }
  }
  render() {
    return (
      <form onSubmit={this.onSubmit}>
        {this.state.error && <p>{this.state.error}</p>}
        <input 
          type='text'
          placeholder='Low'
          autoFocus
          value={this.state.tempLow}
          onChange={this.onLowChange}
        />
        <input 
          type='text'
          placeholder='Hi'
          value={this.state.tempHi}
          onChange={this.onHighChange}
        />
        <input 
          type='text'
          placeholder='Chance of Rain'
          value={this.state.precipProbability}
          onChange={this.onPrecipChange}
        />
        <SingleDatePicker 
          date={this.state.commuteDay} // momentPropTypes.momentObj or null
          onDateChange={this.onDayChange} // PropTypes.func.isRequired
          focused={this.state.calendarFocused} // PropTypes.bool
          onFocusChange={this.onFocusChange} // PropTypes.func.isRequired
          numberOfMonths={1}
          isOutsideRange={this.isOutsideRange} // PropTypes.func
        />
        <Timepicker
          showSecond={false}
          defaultValue={this.state.commuteTime} // momentPropTypes.momentObj or null
          format={'h:mm a'} // See https://momentjs.com/docs/#/parsing/string-format/
          onChange={this.onTimeChange} 
          use12Hours // PropTypes.bool, followed implemenation at http://react-component.github.io/time-picker/examples/12hours.html
        />
        <div>
          <button>Submit</button>
        </div>
      </form>
    );
  }
}