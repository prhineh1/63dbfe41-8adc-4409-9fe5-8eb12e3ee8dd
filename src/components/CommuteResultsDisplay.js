import React from 'react';
import ReactAnimatedWeather from 'react-animated-weather';

const CommuteResultsDisplay = (props) => {
  const modeDeclaration = (
    props.bike ? "You're all clear; bike it" :
    "You should take the metro."
  );
  return (
    <div>
      <h4>{modeDeclaration}</h4>
      <ReactAnimatedWeather
        icon={props.weather.icon}
      />
      <span>
        {props.weather.summary} at 
        {props.weather.temperature} with a 
        {props.weather.precipProbability * 100}% chance of precipitation
      </span>
    </div>
  );
};

export default CommuteResultsDisplay;