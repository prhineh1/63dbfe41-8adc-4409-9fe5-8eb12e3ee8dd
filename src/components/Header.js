import React from 'react';

const Header = () => (
    <header className='header'>
        <div className='content-container'>
            <div className='header__content'>
                <div className='header__content__image' />
            </div>
        </div>
    </header>
);

export default Header;

