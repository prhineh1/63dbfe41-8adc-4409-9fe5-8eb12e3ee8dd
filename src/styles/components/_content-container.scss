.content-container {
    margin: 0 auto;
    padding: 0 $m-size;
    max-width: 80rem;
}

.content-container--body {
    margin: $m-size;
    background: $lavender; 
    border-radius: $m-size;    

    @media (min-width: $desktop-breakpoint) {
        background: $lavender;
        border-radius: $m-size;
        margin: $xl-size auto 0 auto;
    }
}