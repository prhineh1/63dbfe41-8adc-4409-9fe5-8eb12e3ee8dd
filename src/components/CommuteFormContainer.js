import React from 'react';
import { connect } from 'react-redux';
import CommuteForm from './CommuteForm';
import { addUserData } from '../actions/weather';

export class CommuteFormContainer extends React.Component {
  onSubmit = (userData) => {
    this.props.addUserData(userData);
  }
  render() {
    return (
      <div>
        <CommuteForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addUserData: (userData) => dispatch(addUserData(userData))
});

export default connect(null, mapDispatchToProps)(CommuteFormContainer);
