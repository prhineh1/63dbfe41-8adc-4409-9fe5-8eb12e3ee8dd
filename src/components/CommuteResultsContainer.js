import React from 'react';
import { connect } from 'react-redux';
import CommuteResultsDisplay from './CommuteResultsDisplay';

export const CommuteResultsContainer = (props) => {
  return (
    <CommuteResultsDisplay weather={props.weather} />
  ); 
}
const mapStateToProps = (state) => {
  return {
    weather: state.weather
  };
};

export default connect(mapStateToProps)(CommuteResultsContainer);