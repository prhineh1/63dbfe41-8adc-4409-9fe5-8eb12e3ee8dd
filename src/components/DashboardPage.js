import React from 'react';
import CommuteFormContainer from './CommuteFormContainer';


const DashboardPage = () => (
    <div>
        <CommuteFormContainer />
    </div>
);

export default DashboardPage;