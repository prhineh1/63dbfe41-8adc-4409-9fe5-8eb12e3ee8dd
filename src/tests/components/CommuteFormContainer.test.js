import React from 'react';
import { shallow } from 'enzyme';
import { CommuteFormContainer } from '../../components/CommuteFormContainer';
import data from '../fixtures/weatherData';

let wrapper, addUserData;

beforeEach(() => {
  addUserData = jest.fn();
  wrapper = shallow(<CommuteFormContainer addUserData={addUserData} />);
});

test('should render CommuteFormContainer', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle addUserData', () => {
  wrapper.find('CommuteForm').prop('onSubmit')(data[0]);
  expect(addUserData).toHaveBeenLastCalledWith(data[0]);
});
