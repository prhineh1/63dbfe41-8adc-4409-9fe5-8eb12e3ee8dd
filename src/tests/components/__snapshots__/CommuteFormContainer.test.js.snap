// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CommuteFormContainer 1`] = `
<div>
  <CommuteForm
    onSubmit={[Function]}
  />
</div>
`;
